content/200-concepts/100-components/02-prisma-client/060-full-text-search.mdx

```diff
@@ -0,0 +1,93 @@
+---
+title: 'Full-Text Search'
+metaTitle: 'Full-Text Search'
+metaDescription: 'This page explains how to search for text within a field.'
+---
+
+# Full-Text Search
+
+The Prisma Client supports searching for text within a field. You can use full-text search to add search to your application.
+
+The Prisma Client supports a full-text search API for Postgres. With full-text search, you can search for text within a column.
+
+The Full-Text Search API is still in **Preview**. You can enable `fullTextSearch` in your Prisma Schema:
+
+```jsx
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["fullTextSearch"]
+}
+
+model Post {
+	id     Int    @id @default(autoincrement())
+  title  String @unique
+  body   String
+  status Status
+}
+
+enum Status {
+  Draft
+  Live
+}
+```
+
+After your regenerate your client, you'll see a new `search` field on your `String` fields that you can query on. Here's a few examples:
+
+```jsx
+// All posts that contain the words cat or dog.
+const result = await prisma.posts.findMany({
+  where: {
+    body: {
+      search: 'cat | dog',
+    },
+  },
+})
+
+// All drafts that contain the words cat and dog.
+const result = await prisma.posts.findMany({
+  where: {
+    status: 'Draft',
+    body: {
+      search: 'cat & dog',
+    },
+  },
+})
+```
```

Comment:
```suggestion
### Example of new <inlinecode>search</inlinecode> field

```ts
// All posts that contain the words cat or dog.
const result = await prisma.posts.findMany({
  where: {
    body: {
      search: 'cat | dog',
    },
  },
})

// All drafts that contain the words cat and dog.
const result = await prisma.posts.findMany({
  where: {
    status: 'Draft',
    body: {
      search: 'cat & dog',
    },
  },
})
```
```